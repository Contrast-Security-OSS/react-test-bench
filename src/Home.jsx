import { useEffect } from 'react';
import logo from './logo.svg';
import './Home.css';

function Home() {
  useEffect(() => {
    document.title = 'React Test Bench';
  }, []);

  return (
    <div className="Home">
      <header className="Home-header">
        <img src={logo} className="Home-logo" alt="logo" />
        <p>
          Edit <code>src/Home.jsx</code> and save to reload.
        </p>
        <a
          className="Home-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </a>
      </header>
    </div>
  );
}

export default Home;
