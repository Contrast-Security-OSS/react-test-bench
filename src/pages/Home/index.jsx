import Page from '../Page';
import styles from './index.module.css';
import logo from './logo.svg';

function Home() {
  return (
    <Page className={styles.home}>
      <img src={logo} className={styles.logo} alt="logo" />
      <h1>React Test Bench</h1>
      <p>
        An intentionally vulnerable React app for demonstrating Contrast
        Security analysis and instrumentation.
      </p>
    </Page>
  );
}

export default Home;
