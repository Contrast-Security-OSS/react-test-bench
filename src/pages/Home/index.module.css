.home {
  align-items: center;
  background-color: var(--dark) !important;
  color: var(--white);
  display: flex;
  flex-direction: column;
}

.logo {
  height: 40vmin;
  pointer-events: none;
}

@keyframes logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

@media (prefers-reduced-motion: no-preference) {
  .logo {
    animation: logo-spin infinite 20s linear;
  }
}
