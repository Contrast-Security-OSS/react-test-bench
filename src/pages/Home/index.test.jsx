import { render } from '@testing-library/react';
import React from 'react';
import Home from '.';

test('renders learn react link', () => {
  const { container } = render(<Home />);
  expect(container).toMatchSnapshot();
});
