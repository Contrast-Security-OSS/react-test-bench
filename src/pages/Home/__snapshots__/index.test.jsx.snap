// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div>
  <main
    class="home container"
  >
    <img
      alt="logo"
      class="logo"
      src="logo.svg"
    />
    <h1>
      React Test Bench
    </h1>
    <p>
      An intentionally vulnerable React app for demonstrating Contrast Security.
    </p>
  </main>
</div>
`;
